core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -54,9 +54,14 @@ private[spark] abstract class BlockObjectWriter(val blockId: BlockId) {
   def revertPartialWritesAndClose()
 
   /**
-   * Writes an object.
+   * Writes a key-value pair.
    */
-  def write(value: Any)
+  def write(key: Any, value: Any)
+
+  /**
+   * Notify the writer that a record worth of bytes has been written with writeBytes.
```

Comment:
I think this comment is out of date; there isn't a `writeBytes` method in the general `BlockObjectWriter` interface, only in `DiskBlockObjectWriter`.